Antidote/CallsManager.m

```diff
@@ -0,0 +1,331 @@
+
+        // workaround for deadlock in objcTox
+        // https://github.com/Antidote-for-Tox/objcTox/issues/51
+        [self performSelector:@selector(dismissAndSwitchToActiveViewControllerForCall:) withObject:call afterDelay:0];
+    }
+}
+
+- (void)controllerDidChangeContent:(RBQFetchedResultsController *)controller
+{
+    if (controller == self.allActiveCallsController) {
+        self.currentCall = [[self.allActiveCallsController fetchedObjects] firstObject];
+        [self updateDuration:self.currentCall.callDuration];
+    }
+}
+
+#pragma mark - Updates to Current Call
+
+- (void)updateDuration:(NSTimeInterval)duration
+{
+    if ([self.currentCallViewController isKindOfClass:[ActiveCallViewController class]]) {
+        ActiveCallViewController *activeVC = (ActiveCallViewController *)self.currentCallViewController;
+        activeVC.callDuration = duration;
+    }
+}
+
+- (void)setEnableMicrophone:(BOOL)enableMicrophone
+{
+    self.manager.enableMicrophone = enableMicrophone;
+}
+
+- (BOOL)enableMicrophone
+{
+    return self.manager.enableMicrophone;
+}
+
+#pragma mark - ActiveCallViewController Delegate
+
+- (void)activeCallDeclineButtonPressed:(ActiveCallViewController *)controller
+{
+    [self.manager sendCallControl:OCTToxAVCallControlCancel toCall:self.currentCall error:nil];
+}
+
+- (void)activeCallMicButtonPressed:(ActiveCallViewController *)controller
+{
+    BOOL enable = controller.micSelected;
+    self.manager.enableMicrophone = enable;
+    controller.micSelected = ! enable;
+}
+
+- (void)activeCallSpeakerButtonPressed:(ActiveCallViewController *)controller
+{
+    NSError *error;
+
+    if (controller.speakerSelected) {
+        if ([self.manager sendCallControl:OCTToxAVCallControlUnmuteAudio toCall:self.currentCall error:&error]) {
+            controller.speakerSelected = NO;
+        }
+        else if (error.code == OCTToxAVErrorControlInvaldTransition) {
+            controller.speakerSelected = YES;
+        }
+    }
+    else {
+        if ([self.manager sendCallControl:OCTToxAVCallControlMuteAudio toCall:self.currentCall error:&error]) {
+            controller.speakerSelected = YES;
+        }
+        else if (error.code == OCTToxAVErrorControlInvaldTransition) {
+            controller.speakerSelected = NO;
+        }
+    }
+}
+
+- (void)activeCallDeclineIncomingCallButtonPressed:(ActiveCallViewController *)controller
+{
+    [self.manager sendCallControl:OCTToxAVCallControlCancel toCall:self.pendingIncomingCall error:nil];
+    self.pendingIncomingCall = nil;
+    controller.incomingCallCallerName = nil;
+    controller.createIncomingCallView = NO;
+}
+
+- (void)activeCallAnswerIncomingCallButtonPressed:(ActiveCallViewController *)controller
+{
+    [self.manager answerCall:self.pendingIncomingCall enableAudio:YES enableVideo:NO error:nil];
+    self.pendingIncomingCall = nil;
+    controller.incomingCallCallerName = nil;
+    controller.createIncomingCallView = NO;
```

Comment:
So if you try and answer a call while already in a call, it will automatically pause the current call.